lib/animina_web/live/chat_live.ex

```diff
@@ -161,6 +161,7 @@ defmodule AniminaWeb.ChatLive do
       |> Enum.filter(fn x ->
         find_user_flag_for_a_flag(user.flags_join_assoc, x).color == color
       end)
+      |> Enum.uniq()
```

Comment:
Why do we have to run a `uniq/1` here at all? Shouldn't it be unique right away? Where is the duplication coming from?

Reply:
What I mean: Do we have two entries in the database for the same flag (same color)? Because that should not be happening.

Reply:
Let me have a closer look at our query and get back

Reply:
I have found the bug 
```
defp find_user_flag_for_a_flag(user_flags, flag) do
    Enum.find(user_flags, fn x -> x.flag_id == flag.id end)
  end
```

this function returns duplicates and in this case a user may have the same white and green flags

Reply:
OK, go back if needed. But please avoid code which read like `flag.flag.name` like we had in the past. That is not easy to understand. 

I am pretty sure we can find a clever way to add the `flag.name` information to the `user_flag` resource. So that we could do something like ´current_user.user_flags` and not `current_user.user_flags[0].flag`.

Fire at will.

Reply:
Should I merge this PR or not?

Reply:
No , Let me get at fixing it , in a couple of hours I will have found the best solution

Reply:
@wintermeyer  , for readability , would changing the access point name be better  , so we have  `user.traits.flag.name`